authority/provisioner/scep.go

```diff
@@ -163,6 +165,71 @@ func (c *challengeValidationController) Validate(ctx context.Context, csr *x509.
 	return ErrSCEPChallengeInvalid
 }
 
+type notificationController struct {
+	client   *http.Client
+	webhooks []*Webhook
+}
+
+// newNotificationController creates a new notificationController
+// that performs SCEP notifications through webhooks.
+func newNotificationController(client *http.Client, webhooks []*Webhook) *notificationController {
+	scepHooks := []*Webhook{}
+	for _, wh := range webhooks {
+		if wh.Kind != linkedca.Webhook_NOTIFYING.String() {
+			continue
+		}
+		if !isCertTypeOK(wh) {
+			continue
+		}
+		scepHooks = append(scepHooks, wh)
+	}
+	return &notificationController{
+		client:   client,
+		webhooks: scepHooks,
+	}
+}
+
+func (c *notificationController) Success(ctx context.Context, csr *x509.CertificateRequest, cert *x509.Certificate, transactionID string) error {
+	for _, wh := range c.webhooks {
+		req, err := webhook.NewRequestBody(webhook.WithX509CertificateRequest(csr), webhook.WithX509Certificate(nil, cert)) // TODO(hs): pass in the x509util.Certifiate too?
+		if err != nil {
+			return fmt.Errorf("failed creating new webhook request: %w", err)
+		}
+		req.X509Certificate.Raw = cert.Raw // adding the full certificate DER bytes
+		req.SCEPTransactionID = transactionID
+		resp, err := wh.DoWithContext(ctx, c.client, req, nil)
+		if err != nil {
+			return fmt.Errorf("failed executing webhook request: %w", err)
+		}
+		if !resp.Allow { // TODO(hs): different response for notifying?
+			return ErrSCEPNotificationFailed // return early
+		}
```

Comment:
Ok, but the use of this Allow is quite weird. I don't see a use case for a webhook returning 2xx and Allow set to false.

Reply:
I've changed this to not rely on the response body at all. Any OK-ish response will now result in the notification succeeding, based on the existing webhook execution logic.